lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -386,13 +398,19 @@ void SingleExitLoopTransformer::initialize() {
     }
   }
 
+  escapingValueSubstMap = computeEscapingValues();
+}
+
+llvm::DenseMap<SILValue, SILValue>
+SingleExitLoopTransformer::computeEscapingValues() const {
```

Comment:
nit: consider passing llvm::DenseMap<SILValue, SILValue> as an output (pointer) param into the func.

Reply:
I am conflicted here. :) I generally prefer the functional style as it makes it easy to read code, but llvm and swift seems to mostly have inout param style. Is that the preferred style? If so, I will change it. 

This should be as efficient as the compiler will optimize away copies with Return Value Optimization.

Reply:
I will leave it as is then. (Even if I pass an argument, I will need to add the inout parameter to the capture?)

Reply:
if you pass the input by pointer/address into computeEscapingValues(), you can then pass that pointer to the closure by value. but it's not a big deal.